pkg/algorithm/hived_algorithm.go

```diff
@@ -716,74 +716,58 @@ func (h *HivedAlgorithm) processSchedulingRequest(
+	physicalPlacement := groupPhysicalPlacement{}
 	for _, podGpuNum := range gpuNums {
 		podPlacements := virtualPlacement[podGpuNum]
 		physicalPlacement[podGpuNum] = make([]CellList, len(podPlacements))
 		for i, podGpus := range podPlacements {
 			physicalPlacement[podGpuNum][i] = make(CellList, len(podGpus))
 			for j, gpu := range podGpus {
 				vGpu := gpu.(*VirtualCell)
-				if vGpu.GetPhysicalCell() != nil {
-					if groupToPreempt := vGpu.GetPhysicalCell().GetAffinityGroup(); groupToPreempt.lazyPreemptionEnable {
+				if pGpu := vGpu.GetPhysicalCell(); pGpu != nil && pGpu.GetState() == cellUsed {
+					if groupToPreempt := vGpu.GetPhysicalCell().GetUsingGroup(); groupToPreempt.lazyPreemptionEnable {
 						h.lazyPreemptAffinityGroup(groupToPreempt, sr.affinityGroupName)
 					}
 				}
-				pac := vGpu.GetPreAssignedCell()
-				// check if the preassigned cell has been (temporarily) bound to a physical cell
-				preassignedPhysical := pac.GetPhysicalCell()
-				if preassignedPhysical == nil {
-					preassignedPhysical = pac.GetPreBoundPhysicalCell()
-				}
-				if preassignedPhysical == nil {
-					// allocate a new physical cell to the preassigned cell. input a copy of the free cell list
-					// because during the scheduling we should not make in-place change to the data structures
-					// (they can be modified only when adding or deleting pods)
-					c := buddyAlloc(h.getTmpFreeCellList(sr.chain), pac.GetLevel(), suggestedNodes)
-					if c == nil {
-						panic(fmt.Sprintf(
-							"VC Safety Broken: Cannot find physical cell for a VC cell: %v", pac.GetAddress()))
-					} else {
-						preassignedPhysical = c
-						// create binding (which is temporary and will be cleared after the scheduling,
-						// same reason as above)
-						pac.SetPreBoundPhysicalCell(preassignedPhysical)
-						preassignedPhysical.SetPreBoundVirtualCell(pac)
-					}
-				}
-				physicalPlacement[podGpuNum][i][j] = mapNonPreassignedCellToPhysical(vGpu, suggestedNodes)
+				physicalPlacement[podGpuNum][i][j] = mapVirtualCellToPhysical(vGpu, h.freeCellList[sr.chain], suggestedNodes)
 			}
 		}
 	}
+	// the cell bindings are temporary, and should be cleared after the scheduling is done
 	clearPreBindings(virtualPlacement)
-	return physicalPlacement, virtualPlacement
+	return physicalPlacement
 }
 
 // scheduleOpportunisticAffinityGroup calls the opportunistic pod scheduler to schedule an affinity group.
 func (h *HivedAlgorithm) scheduleOpportunisticAffinityGroup(
 	sr schedulingRequest,
-	suggestedNodes common.Set) (physicalPlacement groupPhysicalPlacement) {
+	suggestedNodes common.Set) groupPhysicalPlacement {
 
-	physicalPlacement = h.opportunisticSchedulers[sr.chain].Schedule(
+	physicalPlacement := h.opportunisticSchedulers[sr.chain].Schedule(
 		sr.affinityGroupPodNums, opportunisticPriority, suggestedNodes)
 	if physicalPlacement == nil {
 		klog.Infof("Insufficient capacity in PC for scheduling request: GPU numbers %v, priority %v, chain %v",
```

Comment:
>Insufficient capacity in PC fo [](start = 14, length = 30)

Align with below: 
Failed in scheduling due to insufficient capacity in PC for scheduling request... ?